test/integration/model_profiles_test.go

```diff
@@ -52,3 +53,47 @@ func TestModelProfiles(t *testing.T) {
 		assert.Equal(t, sysCfg().ResourceProfiles[resourceProfileCPU].NodeSelector, pod.Spec.NodeSelector)
 	}, 2*time.Second, time.Second/10, "Resource profile should be applied to the model Pod object")
 }
+
+// TestModelProfiles tests that resource profiles are applied as expected.
+func TestModelResourceProfileWithUserSetValues(t *testing.T) {
```

Comment:
Is there a test case for resourceProfile adds nodeSelector AND user adds nodeSelector in model?

For example resource PRofile adds sam=true as nodeSelector. The user on model sets nick=true

Then the final created model in K8s should have:
```
nodeSelectors:
  sam: true
  nick:true
```

Reply:
I see. Let's call this out somewhere.

Reply:
Clarified comments in CRD